osu.Game/Screens/OnlinePlay/Multiplayer/Match/Playlist/MultiplayerPlaylist.cs

```diff
@@ -121,28 +110,29 @@ private void updateState()
 
                 firstPopulation = false;
             }
+
+            // As a small optimisation, only the ID is required to match the selected item.
+            PlaylistItem? selectedItem = client.Room == null ? null : new PlaylistItem(new APIBeatmap()) { ID = client.Room.Settings.PlaylistItemId };
```

Comment:
Is this worth it? Can't we just use an actual model here? Like is this such an optimisation that a specific  allocation of a reduced-usability instance is worth it over presumably scanning a very short playlist?

I really don't like placeholder models being shoved into components like this; N months down the line you try to do anything with the model passed to such a component, only to discover it's some placeholder unusable instance that you have to go chase down and replace with the real thing.

Reply:
Probably not. Because `RoomUpdated` is called on every change whatsoever, there's always a feeling in the back of my mind like "hmm we probably don't need to be allocating new mod arrays when other peoples' beatmap availabilities change" (for example).

I tend to take it a little bit far when there's an easy optimisation opportunity like this. I'll change it to use `CurrentPlaylistItem`.